syntax = "proto3";

service Admin {
  rpc SetLogLevel(SetLogLevelRequests) returns (LogLevelState) {};
  rpc ResetLogLevel(LogResetRequest) returns (LogLevelState) {};
  rpc GetLogLevel(GetLogLevelRequest) returns (LogLevelState) {};
}

message SetLogLevelRequests {
  string level = 1;
  optional int64 duration = 2;
}

message LogLevelState {
  string current_level = 1;
  int64 seconds_to_reset_level = 2;
  string permanent_log_level = 3;
}

message LogResetRequest {}

message GetLogLevelRequest {}


